native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java

```diff
@@ -374,53 +372,45 @@ public static void writeByteStreamToOutputStream(Environment env, BObject entity
         BStream byteStream = EntityBodyHandler.getByteStream(entity);
         if (byteStream != null) {
             BObject iteratorObj = byteStream.getIteratorObj();
-            CountDownLatch latch = new CountDownLatch(1);
-            writeContent(env, entity, outputStream, iteratorObj, latch);
-            try {
-                int timeout = 120;
-                boolean countDownReached = latch.await(timeout, TimeUnit.SECONDS);
-                if (!countDownReached) {
-                    throw ErrorCreator.createError(StringUtils.fromString(
-                            "Could not complete byte stream serialization within " + timeout + " seconds"));
-                }
-            } catch (InterruptedException e) {
-                log.warn("Interrupted before completing the content write");
-            }
+            writeContent(env, entity, outputStream, iteratorObj);
         }
     }
 
     private static void writeContent(Environment env, BObject entity, OutputStream outputStream,
-                                     BObject iteratorObj, CountDownLatch latch) {
-        env.getRuntime().invokeMethodAsyncConcurrently(iteratorObj, BYTE_STREAM_NEXT_FUNC, null, null, new Callback() {
-            @Override
-            public void notifySuccess(Object result) {
-                if (result == null) {
-                    entity.addNativeData(ENTITY_BYTE_STREAM, null);
-                    latch.countDown();
-                    return;
-                }
-                if (result instanceof BError error) {
-                    entity.addNativeData(ENTITY_BYTE_STREAM, null);
-                    this.notifyFailure(error);
-                }
-                try {
-                    writeContentPart((BMap) result, outputStream);
-                } catch (Exception e) {
-                    latch.countDown();
-                    throw ErrorCreator.createError(StringUtils.fromString(
-                            "Error occurred while writing the stream content: "
-                                    + MimeUtil.removeJavaExceptionPrefix(e.getMessage())));
-                }
-                writeContent(env, entity, outputStream, iteratorObj, latch);
-            }
+                                     BObject iteratorObj) {
+        try {
+            Object result = env.getRuntime().callMethod(iteratorObj, BYTE_STREAM_NEXT_FUNC,
+                    new StrandMetadata(true, null));
```

Comment:
Here, we are reading content from a ballerina stream and write it back to a java output stream. My understanding is that `new StrandMetadata(true, null)` creates a new strand. In that case should we create a new `StrandMetadata` for each call of the `writeContent` method?

Reply:
Yes. We do not need to pass strand metadata here.